packages/gravity-ui-web/docs/coding-standards.md

```diff
@@ -0,0 +1,299 @@
+    * This makes the HTML (and the rendered DOM in your browser's dev tools) easier to read and navigate
+    * In complex UIs it may also reduce the memory and processing overheads. The fewer DOM nodes there are, the quicker they can be traversed.
+
+
+## SASS / CSS
+### General
+We use SASS (specifically, the SCSS syntax) to author our CSS. Most of the following rules would apply even if you were writing plain CSS, but some are SASS-specific:
+
+* Don't use vendor prefixes.
+    * We may use a CSS post-processor to add some automatically for better compatibility with old browsers, so it is redundant to add them manually in the source code.
+* Prefer to inherit styles from the cascade over explicitly setting them.
+    * We intentionally lean on the cascade as this can lead to more succinct CSS code and make it easier for us to apply far-reaching global style changes.
+    * As a rule of thumb, write your HTML first. See how it looks without any additional CSS and then selectively add or override only the properties you need to.
+    * Browser dev tools can be helpful here as they let you quickly identify redundant overrides.
+* Use only relative length units
+    * `px` values are forbidden. Font sizes, spacing and dimensions should be set using relative units like `em`, `rem` or `%`. Note, that often times [components should _not_ be setting explicit widths or heights on themselves anyway](./layout-and-grid.md#per-component-rules). As much as possible, we want to allow their default behaviour of growing to fill their parent's width and to accommodate their contents.
+    * Using `em` and `rem` for text and media queries also means that our entire UI layout can scale up or down and responsively adapt based on the user's default text size preference.
+* Avoid hard-coded values
+    * Global SASS variables (based on the [`gravity-particles` design tokens](https://github.com/buildit/gravity-particles/)), functions and mixins exist for most values you are likely to need, such as typography values, [colors](./color-system.md), spacing values (for margins and paddings), etc.
+    * Drawing from the global values helps enforce visual consistency in our UIs. It also makes future redesigns or reskins much easier to implement, since we can simply alter those central values.
+    * It also makes the code easier to understand since the variable and function names convey the _purpose_ of the values, whereas the raw values do not.
+
+As much as possible, we use [Stylelint](https://stylelint.io/) to enforce these rules. You can run it anytime via `npm run lint:style`.
+
+### Declaration and property ordering
+Within a CSS block, declarations should be ordered as follows:
+
+```scss
+.grav-foobar {
+  // SASS @extends
+  @extend .grav-other-thing;
+
+  // SASS mixin @includes that do NOT use content blocks
+  @include grav-fancy-mixin(...);
+
+  // CSS custom properties
+  --foobar-barfoo: 42;
+
+  // SASS variable declartions
+  $foobar: 666;
```

Comment:
if we declare sass variables after custom properties how would we assign the sass variable to the custom property in the block?

don't know that there's a use case for it but seems like we could just put them the other way round and not have the problem?

Reply:
Good point! I'll update the rule and doc accordingly!